tracing/tracing.cc

```diff
@@ -180,17 +180,16 @@ future<> tracing::shutdown() {
     write_pending_records();
     _down = true;
     _write_timer.cancel();
-    return _tracing_backend_helper_ptr->shutdown().then([] {
-        tracing_logger.info("Tracing is down");
-    });
+    co_await _tracing_backend_helper_ptr->shutdown();
+    tracing_logger.info("Tracing is down");
 }
 
 future<> tracing::stop() {
     if (!_down) {
         throw std::logic_error("tracing: stop() called before shutdown()");
     }
 
-    return make_ready_future<>();
+    co_return;
```

Comment:
Not much improvement here, but I agree that coroutine should be the default unless there is a reason against (and there is no reason against here).

Reply:
Next patch makes a bit more use of it:

```patch
 future<> tracing::stop() {
-    if (!_down) {
-        throw std::logic_error("tracing: stop() called before shutdown()");
-    }
-
-    co_return;
+    co_await shutdown();
 }
```